src/components/Transaction/TransactionComponent.tsx

```diff
@@ -0,0 +1,199 @@
+import React, { useEffect, useState } from 'react'
+import style from './Transaction.module.scss'
+import { BsCoin } from 'react-icons/bs'
+import { Button, Modal, NumberInput, SegmentedControl } from '@mantine/core'
+import Stripe from 'stripe'
+import { loadStripe } from '@stripe/stripe-js'
+import { Elements } from '@stripe/react-stripe-js'
+import CheckoutForm from './CheckoutForm'
+import { IconCurrencyDollar } from '@tabler/icons-react'
+import { getConversionRateList } from '../../service/TransactionService'
+import ExchangePolicy from './ExchangePolicy'
+import TransactionHistory from './TransactionHistory'
+import CreditComponent from './CreditComponent'
+
+export default function TransactionComponent() {
+  const [conversionRate, getConversionRate] = useState('')
+  const [creditAmount, setCreditAmount] = useState<number | string>(0)
+  const [creditAmountError, setCreditAmountError] = useState('')
+  const [shouldShowCheckoutForm, setShouldShowCheckoutForm] = useState(false)
+  const [shouldUpdate, setShouldUpdate] = useState(false)
+
+  const [clientSecret, setClientSecret] = useState<string>()
+  const [resetPayment, setResetPayment] = useState(false)
+  const stripePromise = loadStripe(
+    process.env.REACT_APP_STRIPE_PUBLISHABLE_KEY!,
+  )
+  const [value, setValue] = useState('Exchange Policy')
+  const [opened, setOpened] = useState(false)
+
+  const [intentId, setIntentId] = useState('')
+
+  const handleBuyCredit = async () => {
+    setOpened(false)
+    setValue('Proceed To Payment')
+    setCreditAmountError('')
+    if (Number(creditAmount) <= 0) {
+      setCreditAmountError('Please enter credit amount')
+      return
+    }
+    if (Number(creditAmount) > 500) {
+      setCreditAmountError('You can only buy up to 500 credits')
+      return
+    }
+
+    setShouldShowCheckoutForm(true)
+
+    const stripe = new Stripe(process.env.REACT_APP_STRIPE_SECRET_KEY!, {
+      apiVersion: '2023-10-16',
+    })
+
+    try {
+      const paymentIntents = await stripe.paymentIntents.create({
```

Comment:
Create a Stripe services file and write into it.

Reply:
fixed